bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureBridgeHandler.java

```diff
@@ -123,15 +123,15 @@ public void initialize() {
                         if (!session.initialize(authstring, pinCode, config.username, config.password)) {
                             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
                                     "Failed to login to Verisure, please check your account settings! Is MFA activated?");
-                            return;
                         }
-                        startAutomaticRefresh(config.refresh);
                     }
-                });
-            } catch (RuntimeException e) {
-                logger.warn("Failed to initialize: {}", e.getMessage());
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            }
+                } catch (RuntimeException e) {
+                    logger.warn("Failed to initialize: {}", e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } finally {
+                    startAutomaticRefresh(config.refresh);
+                }
```

Comment:
Ok, I just have a new look more in details.
I understand your need to refresh and so start the refresh job even if the initial init failed.

Just one comment: why not catching RuntimeException directly in VerisureSession ? session.initialize should then return false. Like that, no need to deal with exception in the thing handler.

Reply:
> I have also found some places where I need to add a Null check in VerisureSession.java.
> Should I add a new PR for that?

Just try to provide consistent PRs. You are the master of this binding, I let you decide if this is better to include the null check in this PR or in another PR.

Reply:
In fact, the real problem is your method `refreshAndUpdateStatus` not catching exception. So in case of exception, this will stop your refresh job for ever.
If `session.refresh()` can throw an exception, you certainly have to add a try/catch in `refreshAndUpdateStatus` to avoid that.

Reply:
I found a much better solution, thanks for your hint, and also added some more null checks.